domain/model/state/controller/types.go

```diff
@@ -0,0 +1,345 @@
+}
+
+func (m *dbModel) toCoreModel() (coremodel.Model, error) {
+	var credOwnerName user.Name
+	if m.CredentialOwnerName != "" {
+		var err error
+		credOwnerName, err = user.NewName(m.CredentialOwnerName)
+		if err != nil {
+			return coremodel.Model{}, errors.Capture(err)
+		}
+	}
+
+	var cloudRegion string
+	if m.CloudRegion.Valid {
+		cloudRegion = m.CloudRegion.String
+	}
+
+	var credentialName string
+	if m.CredentialName.Valid {
+		credentialName = m.CredentialName.String
+	}
+
+	var credentialCloudName string
+	if m.CredentialCloudName.Valid {
+		credentialCloudName = m.CredentialCloudName.String
+	}
+
+	return coremodel.Model{
+		Name:        m.Name,
+		Qualifier:   coremodel.Qualifier(m.Qualifier),
+		Life:        corelife.Value(m.Life),
+		UUID:        coremodel.UUID(m.UUID),
+		ModelType:   coremodel.ModelType(m.ModelType),
+		Cloud:       m.CloudName,
+		CloudType:   m.CloudType,
+		CloudRegion: cloudRegion,
+		Credential: credential.Key{
+			Name:  credentialName,
+			Cloud: credentialCloudName,
+			Owner: credOwnerName,
+		},
+	}, nil
+}
+
+// dbInitialModel represents initial the model state written to the model table on
+// model creation.
+type dbInitialModel struct {
+	// UUID is the universally unique identifier of the model.
+	UUID string `db:"uuid"`
+
+	// CloudUUID is the unique identifier for the cloud the model is on.
+	CloudUUID string `db:"cloud_uuid"`
+
+	// ModelType is the type of model.
+	ModelType string `db:"model_type"`
+
+	// LifeID the ID of the current state of the model.
+	LifeID int `db:"life_id"`
+
+	// Name is the human friendly name of the model.
+	Name string `db:"name"`
+
+	// Qualifier is used to disambiguate the model name.
+	Qualifier string `db:"qualifier"`
+}
+
+type dbNames struct {
+	ModelName      string `db:"name"`
+	ModelQualifier string `db:"qualifier"`
+}
+
+// dbUUID represents a UUID.
+type dbUUID struct {
+	UUID string `db:"uuid"`
+}
+
+// dbModelUUIDRef represents the model uuid in tables that have a foreign key on
+// the model uuid.
+type dbModelUUIDRef struct {
+	ModelUUID string `db:"model_uuid"`
+}
+
+type dbModelUUID struct {
+	UUID string `db:"uuid"`
+}
+
+type dbModelNameAndQualifier struct {
+	// Name is the model name.
+	Name string `db:"name"`
+	// Qualifier is the model qualifier.
+	Qualifier string `db:"qualifier"`
+}
+
+// dbModelType represents the model type from the model table.
+type dbModelType struct {
+	Type string `db:"type"`
+}
+
+// dbUserModelSummary is summary of the information for a model from the
+// perspective of a user that has access to the model.
+type dbUserModelSummary struct {
+	// Access is the access level the supplied user has on this model.
+	Access permission.Access `db:"access_type"`
+
+	// UserLastConnection is the last time this user has accessed this model.
+	UserLastConnection *time.Time `db:"time"`
+
+	// Life is the current model's life value.
+	Life string `db:"life"`
+
+	// Model state related members
```

Comment:
I don't think these provider any value (I know these are just relocations).

Reply:
There is a lot of this about. I'll follow up.